### ISSUE TITLE
Broken resolvers config in apigateway container

### ISSUE THREAD

**Issue by lornajane** [apache/openwhisk#3428]
## Environment details:

* Attempting to do local ubuntu setup but it's not going well ...  
* version of docker: `17.12.1-ce`, ansible `2.3.0`, ubuntu: `17.10`

## Steps to reproduce the issue:

1.   Follow ubuntu setup instructions
2.   At the point where `ansible-playbook apigateway.yml` is next command, observe that it times out checking that the gateway comes up.

## Provide the expected results and outputs:
Ansible playbook completes successfully

## Provide the actual results and outputs:

When the playbook starts checking if the API Gateway is up, you need to _very quickly_ do
```
docker exec -it apigateway /bin/bash
vi /etc/apigateway/conf.d/includes/resolvers.conf // (remove the word "run" from this file, save and exit)
/usr/local/sbin/api-gateway
```

... the ansible playbook will then find the API Gateway is up and proceed

## Additional information you deem important:
Happens repeatably, the line that needs editing seems to be in an image rather than in our project codebase.

**Comment by mdeuser** [comment]
@lornajane - what was the before/after contents of the `resolvers.conf` file that you updated.

in my local environment running apigateway:0.9.8, that file looks like
```
bash-4.3# cat resolvers.conf
resolver 10.0.2.3;
```

**Comment by lornajane** [comment]
It's like `resolvers run 8.8.8.8, 8.8.4.4` I think, I don't seem to be able to connect to the container at all now.  I removed the `run` and then I could start the API gateway.

**Comment by Jbalkind** [comment]
I have been seeing this same problem, running OpenWhisk in an Ubuntu 18.04 VM in Qemu. I figured out what's happening and it's not upstream alpine's fault.

The `/etc/api-gateway/conf.d/includes/resolvers.conf` file is creaetd by the incubator-openwhisk-apigateway `init.sh` script, in particular the following line ([see inline](https://github.com/apache/incubator-openwhisk-apigateway/blob/master/init.sh#L58)):

```
echo resolver $(awk 'BEGIN{ORS=" "} /nameserver/{print $2}' /etc/resolv.conf | sed "s/ $/;/g") > /etc/api-gateway/conf.d/includes/resolvers.conf
```

At the face of it, this seems correct. The problem is that there are comments at the start of the file. They look like this on my system:

```
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.
```

The last line of the comment matches the awk command as it contains `nameserver`. Then, as the awk command says, it prints the 2nd token on the line, which is the word `run`.

This can be simply solved by changing the awk command to:
```
awk 'BEGIN{ORS=" "} /^nameserver/{print $2}' /etc/resolv.conf
```
or, for something more robust (in case there's space before `nameserver`), you could go for:
```
awk 'BEGIN{ORS=" "} /^[[:space:]]*nameserver/{print $2}' /etc/resolv.conf
```

**Comment by Jbalkind** [comment]
Actually looking on the most recent apigateway, this has been fixed. My system was pointed to the old commit. It should be possible to close this now, I think.

**Comment by Jbalkind** [comment]
The most recent apigateway fixes this (as of commit 2b873660d1931d504b9e17405fb99630533be6da ) but this repository has not been updated to point to a newer docker file for the apigateway.

This can be fixed by changing `version` for apigateway in `ansible/group_vars/all` (see [here](https://github.com/apache/incubator-openwhisk/blob/master/ansible/group_vars/all#L251)) to either `0.9.12` or `latest`.
